controllers/reconcile_configmap.go

```diff
@@ -0,0 +1,144 @@
+
+	nropv1alpha1 "github.com/openshift-kni/numaresources-operator/api/numaresourcesoperator/v1alpha1"
+	"github.com/openshift-kni/numaresources-operator/pkg/apply"
+	cfgstate "github.com/openshift-kni/numaresources-operator/pkg/objectstate/cfg"
+	rtestate "github.com/openshift-kni/numaresources-operator/pkg/objectstate/rte"
+	rteconfig "github.com/openshift-kni/numaresources-operator/rte/pkg/config"
+	"github.com/openshift-kni/numaresources-operator/rte/pkg/sysinfo"
+)
+
+const (
+	ConfigDataField = "config.yaml"
+)
+
+type configMapHelper struct {
+	cli       client.Client
+	scm       *runtime.Scheme
+	instance  *nropv1alpha1.NUMAResourcesOperator
+	namespace string
+}
+
+func (cmh *configMapHelper) reconcileConfigMap(ctx context.Context, kcKey client.ObjectKey) (*corev1.ConfigMap, error) {
+	mcoKc := &mcov1.KubeletConfig{}
+	if err := cmh.cli.Get(ctx, kcKey, mcoKc); err != nil {
+		return nil, err
+	}
+
+	kubeletConfig, err := mcoKubeletConfToKubeletConf(mcoKc)
+	if err != nil {
+		return nil, err
+	}
+
+	mcps, err := getNodeGroupsMCPs(ctx, cmh.cli, cmh.instance.Spec.NodeGroups)
+	if err != nil {
+		return nil, err
+	}
+
+	mcp, err := findMCPForKubeletConfig(mcps, mcoKc)
+	if err != nil {
+		return nil, err
+	}
+
+	generatedName := rtestate.GetComponentName(cmh.instance.Name, mcp.Name)
+	return cmh.syncConfigMap(ctx, kubeletConfig, generatedName)
+}
+
+func (cmh *configMapHelper) syncConfigMap(ctx context.Context, kubeletConfig *kubeletconfigv1beta1.KubeletConfiguration, name string) (*corev1.ConfigMap, error) {
+	rendered, err := renderRTEConfig(cmh.namespace, name, kubeletConfig)
+	if err != nil {
+		klog.ErrorS(err, "rendering config", "namespace", cmh.namespace, "name", name)
+		return nil, err
+	}
+
+	cfgManifests := cfgstate.Manifests{
+		Config: rendered,
+	}
+	existing := cfgstate.FromClient(context.TODO(), cmh.cli, cmh.namespace, name)
+	for _, objState := range existing.State(cfgManifests) {
+		if err := controllerutil.SetControllerReference(cmh.instance, objState.Desired, cmh.scm); err != nil {
+			return nil, errors.Wrapf(err, "Failed to set controller reference to %s %s", objState.Desired.GetNamespace(), objState.Desired.GetName())
+		}
+		if _, err := apply.ApplyObject(context.TODO(), cmh.cli, objState); err != nil {
+			return nil, errors.Wrapf(err, "could not create %s", objState.Desired.GetObjectKind().GroupVersionKind().String())
+		}
+	}
+	return rendered, nil
+}
+
+func findMCPForKubeletConfig(mcps []*machineconfigv1.MachineConfigPool, mcoKc *mcov1.KubeletConfig) (*machineconfigv1.MachineConfigPool, error) {
+	for _, mcp := range mcps {
+		if mcoKc.Spec.MachineConfigPoolSelector == nil {
+			continue
+		}
+
+		selector, err := metav1.LabelSelectorAsSelector(mcoKc.Spec.MachineConfigPoolSelector)
+		if err != nil {
+			return nil, err
+		}
+
+		if selector.Matches(labels.Set(mcp.Labels)) {
+			return mcp, nil
+		}
+	}
+	return nil, fmt.Errorf("no match")
```

Comment:
we can use a more elaborate error message here

Reply:
yeah, I'll be handling on a separate PR